src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -315,50 +365,64 @@ public bool TryPeekOne(out MessageEnvelope envelope, TimeSpan? max = null)
         public bool TryPeekOne(out MessageEnvelope envelope, TimeSpan? max, CancellationToken cancellationToken)
         {
             return InternalTryPeekOne(out envelope, max, cancellationToken, true);
-        }        
+        }
+
+        /// <inheritdoc cref="TryPeekOne(out MessageEnvelope, TimeSpan?, CancellationToken)"/>
+        public async ValueTask<(bool success, MessageEnvelope envelope)> TryPeekOneAsync(TimeSpan? max, CancellationToken cancellationToken)
+        {
+            return await InternalTryPeekOneAsync(max, cancellationToken, true).ConfigureAwait(false);
+        }
 
         private bool InternalTryPeekOne(out MessageEnvelope envelope, TimeSpan? max, CancellationToken cancellationToken, bool shouldLog)
         {
-            bool didPeek;
+            var received = InternalTryPeekOneAsync(max, cancellationToken, shouldLog).GetAwaiter().GetResult();
+            envelope = received.envelope;
+            return received.success;
+        }
+        private async ValueTask<(bool success, MessageEnvelope envelope)> InternalTryPeekOneAsync(TimeSpan? max, CancellationToken cancellationToken, bool shouldLog)
+        {
+            (bool didPeek, MessageEnvelope env) peek;
             var maxDuration = GetTimeoutOrDefault(max);
             var start = Now;
-            
+
             if (maxDuration.IsZero())
             {
                 ConditionalLog(shouldLog, "Trying to peek message from TestActor queue. Will not wait.");
-                didPeek = _testState.Queue.TryPeek(out envelope);
+                peek = await _testState.Queue.TryPeekAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This case requires the test kit to return immediately with a value, regardless of the queue state. Don't use the async version of this method, use `TryPeek` instead.